Mathlib/Analysis/SpecialFunctions/Complex/Arg.lean

```diff
@@ -512,6 +515,20 @@ theorem arg_coe_angle_eq_iff {x y : ℂ} : (arg x : Real.Angle) = arg y ↔ arg
   simp_rw [← Real.Angle.toReal_inj, arg_coe_angle_toReal_eq_arg]
 #align complex.arg_coe_angle_eq_iff Complex.arg_coe_angle_eq_iff
 
+lemma arg_mul' {x y : ℂ} {a₁ a₂ b₁ b₂ : ℝ} (h₁ : -Real.pi ≤ a₁ + b₁) (h₂ : a₂ + b₂ ≤ Real.pi)
+    (hx₀ : x ≠ 0) (hx₁ : a₁ < x.arg) (hx₂ : x.arg ≤ a₂)
+    (hy₀ : y ≠ 0) (hy₁ : b₁ < y.arg) (hy₂ : y.arg ≤ b₂) :
+    (x * y).arg = x.arg + y.arg := by
+  rw [← arg_coe_angle_toReal_eq_arg, arg_mul_coe_angle hx₀ hy₀, ← Real.Angle.coe_add,
+      Real.Angle.toReal_coe_eq_self_iff_mem_Ioc]
+  exact ⟨by linarith, by linarith⟩
+
+lemma arg_mul {x y : ℂ} (hx₀ : x ≠ 0) (hx₁ : -Real.pi / 2 < x.arg) (hx₂ : x.arg ≤ Real.pi / 2)
+    (hy₀ : y ≠ 0) (hy₁ : -Real.pi / 2 < y.arg) (hy₂ : y.arg ≤ Real.pi / 2) :
+    (x * y).arg = x.arg + y.arg :=
+  arg_mul' (a₁ := -Real.pi / 2) (a₂ := Real.pi / 2) (b₁ := -Real.pi / 2) (b₂ := Real.pi / 2)
+    (by linarith) (by linarith) hx₀ hx₁ hx₂ hy₀ hy₁ hy₂
```

Comment:
My inclination is that the main version of this lemma should actually have the inequality hypotheses in terms of `arg x + arg y`, e.g. `arg x + arg y ∈ Set.Ioc (-π) π`, and then callers with separate inequalities on each argument can use `⟨by linarith, by linarith⟩` themselves as needed. In fact that more general version is true as an `iff`, though the other direction of the `iff` is probably rather less often useful.

Reply:
Good point. Changing to
```lean
lemma arg_mul_eq_add_arg_iff {x y : ℂ} (hx₀ : x ≠ 0) (hy₀ : y ≠ 0) :
    (x * y).arg = x.arg + y.arg ↔ arg x + arg y ∈ Set.Ioc (-π) π := by
  rw [← arg_coe_angle_toReal_eq_arg, arg_mul_coe_angle hx₀ hy₀, ← Real.Angle.coe_add,
      Real.Angle.toReal_coe_eq_self_iff_mem_Ioc]

lemma arg_mul {x y : ℂ} (hx₀ : x ≠ 0) (hx₁ : -Real.pi / 2 < x.arg) (hx₂ : x.arg ≤ Real.pi / 2)
    (hy₀ : y ≠ 0) (hy₁ : -Real.pi / 2 < y.arg) (hy₂ : y.arg ≤ Real.pi / 2) :
    (x * y).arg = x.arg + y.arg :=
  (arg_mul_eq_add_arg_iff hx₀ hy₀).mpr (by rw [Set.mem_Ioc]; constructor <;> linarith)
```
and analogously for `log`.

Reply:
Is more to be done here?

Reply:
I think it should instead be:
```
alias ⟨_, arg_mul⟩ := arg_mul_eq_add_arg_iff
```
As Johan said, the caller can simply use `linarith` to close the goals if they have the more specific conditions on the angles. Same goes for `log_mul`

Reply:
OK; done.